app/controllers/content_items_controller.rb

```diff
@@ -56,6 +56,11 @@ def set_guide_draft_access_token
 
   def load_content_item
     content_item = Services.content_store.content_item(content_item_path)
+
+    if Services.feature_toggler.use_recommended_related_links?(content_item['links'], request.headers)
+      content_item['links']['ordered_related_items'] = content_item['links'].fetch('suggested_ordered_related_items', [])
```

Comment:
I think you need to set a response header here to avoid the two variants being mixed up in the cache.  Have you seen https://github.com/alphagov/govuk_ab_testing?  It does that and also some analytics stuff.

Reply:
We're not using the govuk_ab_testing gem here purposefully, as this isn't a test but instead a mechanism to achieve rollback if we ever need to (for example, if we realise that some proportion of generated links are in some way bad). We'll be setting the request header in the CDN which will effectively invalidate the cache if we change this header, which means we shouldn't need to set a response header (again, we're only ever going to be showing one of these 'variants' at any time).